comfyui_manager/legacy/share_3rdparty.py

```diff
@@ -0,0 +1,386 @@
+            form.add_field("sharedWorkflowPromptJsonString", json.dumps(prompt['output']))
+            form.add_field("shareWorkflowCredits", credits)
+            form.add_field("shareWorkflowTitle", title)
+            form.add_field("shareWorkflowDescription", description)
+            form.add_field("shareWorkflowIsNSFW", str(is_nsfw).lower())
+            form.add_field("currentSnapshot", json.dumps(await core.get_current_snapshot()))
+            form.add_field("modelsInfo", json.dumps(models_info))
+
+            async with session.post(
+                    f"{share_endpoint}/upload_workflow",
+                    data=form,
+            ) as resp:
+                assert resp.status == 200
+                upload_workflow_json = await resp.json()
+                workflowId = upload_workflow_json["workflowId"]
+
+    # check if the user has provided Matrix credentials
+    if "matrix" in share_destinations:
+        comfyui_share_room_id = '!LGYSoacpJPhIfBqVfb:matrix.org'
+        filename = os.path.basename(asset_filepath)
+        content_type = assetFileType
+
+        try:
+            from matrix_client.api import MatrixHttpApi
+            from matrix_client.client import MatrixClient
+
+            homeserver = 'matrix.org'
+            if matrix_auth:
+                homeserver = matrix_auth.get('homeserver', 'matrix.org')
+            homeserver = homeserver.replace("http://", "https://")
+            if not homeserver.startswith("https://"):
+                homeserver = "https://" + homeserver
+
+            client = MatrixClient(homeserver)
+            try:
+                token = client.login(username=matrix_auth['username'], password=matrix_auth['password'])
+                if not token:
+                    return web.json_response({"error": "Invalid Matrix credentials."}, content_type='application/json', status=400)
+            except:
+                return web.json_response({"error": "Invalid Matrix credentials."}, content_type='application/json', status=400)
+
+            matrix = MatrixHttpApi(homeserver, token=token)
+            with open(asset_filepath, 'rb') as f:
+                mxc_url = matrix.media_upload(f.read(), content_type, filename=filename)['content_uri']
+
+            workflow_json_mxc_url = matrix.media_upload(prompt['workflow'], 'application/json', filename='workflow.json')['content_uri']
+
+            text_content = ""
+            if title:
+                text_content += f"{title}\n"
+            if description:
+                text_content += f"{description}\n"
+            if credits:
+                text_content += f"\ncredits: {credits}\n"
+            matrix.send_message(comfyui_share_room_id, text_content)
+            matrix.send_content(comfyui_share_room_id, mxc_url, filename, 'm.image')
+            matrix.send_content(comfyui_share_room_id, workflow_json_mxc_url, 'workflow.json', 'm.file')
+        except:
```

Comment:
this will catch `KeyboardInterrupt` and `SystemExit`, do not use if it is intended (_adding `Exception` would be nice.)_

`logging.exception("An error occurred")` line replaces next two lines with `traceback`.

_please ignore my comments if this is how it was intended - it just caught my eye when I opened PR in the morning with a cup of coffee =)_